### DISCUSSION TITLE
Long G-code String Problem

### DISCUSSION THREAD

**Discussion by UD551** [supermerill/SuperSlicer#1431] [Q&A] [answered]
Hello, I am facing a problem when using a HYREL 3D printer.
I have used Super Slicer for a while now and I consider myself to be good at using it. However, a recent error occurred when I tried printing a 7 hr print. The printer stopped because of an" Invalid long g-code error". Which is basically long lines of code that are being generated. Even though, these codes are commented out. I still can't figure out a way to remove them using Super Slicer rather than editing the G-code manually.

I hope you will be able to help me out! Thank you in advance for any guidance possible. 

Some examples of the long G-code string error which are shown on the HYREL  error window are: 
" 
11:27:48 AM  
Invalid long Gcode string found
; start_gcode = G1 H0 X0.01; Clears H2 from previous offset.\nG28 Z0 ; home Z axes\nG28 X0 Y0 ; home X and Y axes\nG53 ; clear fixture offsets\n;G1 Z5 F5000 ; lift nozzle\nM660 H2 Z35.504; (-Perfect 6 inch Glass-)\n;G54 X106.50 Y60.50; define center offset (print bed origin)\nM229 E1 D1 S0.02; Enable Evalues and disables prime/unprime for non-print moves under 0.02seconds\n;M106 C255 ; run fan 0-255 instead of 0-100 (now accounted for with check in slicer settings)\nM795 S015; Sets the fine Z adjust in pulses 3pulses=4um  1333pulses per mm or 1.3 pulses per micron\n\nM190 S[bed_temperature] ; wait for bed temp\nM109 S[first_layer_temperature] T12 ; wait for head 2 to reach temp\n\nG1 X0.1 H2\n\n;M721 S10000 E00 P0 T0\n\nT1 ; force second slot\n\n;M722 S10000 E00 P0 T0

1:30:30 PM  
Invalid long Gcode string found
; before_layer_gcode = ;announce new layer <[layer_num]>\n;---\n;M756 S[layer_height]\nM790 ;execute any new layer actions\nM106 S100 T11 ;P100 just for while extruding\n;---

1:30:47 PM  
Invalid long Gcode string found
; end_gcode = M104 S0 ; turn off temperature\nG28 X0  ; home X axis\nM84     ; disable motors\nG53 ; clear fixture offsets\nG1 H0 X0.01; Clears H2 from previous offset.\nM30 ; end job\n

1:31:05 PM  
Invalid long Gcode string found
; filament_ramming_parameters = "120 100 6.6 6.8 7.2 7.6 7.9 8.2 8.7 9.4 9.9 10.0| 0.05 6.6 0.45 6.8 0.95 7.8 1.45 8.3 1.95 9.7 2.45 10 2.95 7.6 3.45 7.6 3.95 7.6 4.45 7.6 4.95 7.6";"120 100 6.6 6.8 7.2 7.6 7.9 8.2 8.7 9.4 9.9 10.0| 0.05 6.6 0.45 6.8 0.95 7.8 1.45 8.3 1.95 9.7 2.45 10 2.95 7.6 3.45 7.6 3.95 7.6 4.45 7.6 4.95 7.6"

1:31:29 PM  
Invalid long Gcode string found
; printer_notes = Using Heads 2 (HT MK1 with .25mm nozzle) and \nHead 4 (SDS10)\nHead 3 is and SDS 5 for electrospinning not fully setup.\nLast edited 2021 06 16 by RKP

1:31:30 PM  
Invalid long Gcode string found
; start_gcode = G1 H0 X0.01; Clears H2 from previous offset.\nG28 Z0 ; home Z axes\nG28 X0 Y0 ; home X and Y axes\nG53 ; clear fixture offsets\n;G1 Z5 F5000 ; lift nozzle\nM660 H2 Z35.504; (-Perfect 6 inch Glass-)\n;G54 X106.50 Y60.50; define center offset (print bed origin)\nM229 E1 D1 S0.02; Enable Evalues and disables prime/unprime for non-print moves under 0.02seconds\n;M106 C255 ; run fan 0-255 instead of 0-100 (now accounted for with check in slicer settings)\nM795 S015; Sets the fine Z adjust in pulses 3pulses=4um  1333pulses per mm or 1.3 pulses per micron\n\nM190 S[bed_temperature] ; wait for bed temp\nM109 S[first_layer_temperature] T12 ; wait for head 2 to reach temp\n\nG1 X0.1 H2\n\n;M721 S10000 E00 P0 T0\n\nT1 ; force second slot\n\n;M722 S10000 E00 P0 T0
"v]

**Comment by neophyl** [comment, marked as answer]
Those look like your start/end gcode blocks.  If you open up the output gcode at the end of the gcode after all the print moves etc Slicer/PS/Super Slicer stores a copy of all the settings used.  Its a pretty handy function that allows you to load /import all the settings from any output gcode.
Your printer firmware should NOT be parsing those as they are after the end and besides they are commented out.  Sounds like the firmware on the printer could do with a bug fix.  As Slicer has always done this with gcode files it seems strange that its only recently started though.

As this is causing you an issue I can see a couple of possible ways to address it.  
-Open an issue here and supply the project and output gcode in the hopes that Merill can figure out why firmware on a printer he doesnt have access to is having a problem. Adding a switch option to not include the settings into the output may be a valid way of dealing with it (although personally finding the reason why its started happening now would still be bugging me :)  )
-Run a post process script to automatically remove the entries after the last print move.  Basically automate your manual editing process.

**Reply by UD551** [reply]
Yes you are right! I think it is a problem from the software that the printer is using. 

I will try to use the post processing script and it might work. 
Thank you so much on the insight and help.
Have a great day.
